### ISSUE TITLE
Poor Image Quality after crop

### ISSUE THREAD

**Issue by imbudhiraja** [ivpusic/react-native-image-crop-picker#1085]
### Version
- react-native-image-crop-picker v0.23.1
- react-native v0.59.4

### Platform
- iOS
- Android

### Steps to reproduce

1. Open Gallery
2. Select Image
3. Crop Image

You will get a poor image quality after cropping.

Reproduce-able code 
```
const config = {
  cropperStatusBarColor: 'black',
  cropperToolbarColor: 'black',
  cropping: true,
  mediaType: 'photo',
  showCropFrame: true,
  showCropGuidelines: true,
};

ImagePicker.openCamera(config)
  .then((response) => {})
  .catch((err) => {});

ImagePicker.openPicker(config)
  .then((response) => {})
  .catch((err) => {});

```

Love react-native-image-crop-picker? Please consider supporting our collective:
👉  https://opencollective.com/react-native-image-crop-picker/donate

**Comment by SourceCipher** [comment]
@imbudhiraja What are your dimensions of the crop? 

For me I want to set to 150x150 images and it seems to be an awful quality. If I set to convert some 50x50 for the profile pics, then the quality is impossible to have. I found anything more than 500x500 turns out a good one but the size to store isnt

**Comment by imbudhiraja** [comment]
@SourceCipher I was using 320*120 for cover pictures

**Comment by SourceCipher** [comment]
I have tested both with base64 and original storage image and they all come up in a very bad quality. I uploaded them to the db and they are just unusable and the dimensions of my pic was 300 x 300.

I have checked issues on this library and a lot of them are with bad quality images dating back to 2017. Yet nobody managed to fix it as its the main part of the library. No point having it if the image quality turns out unusable.

**Comment by caetius** [comment]
Any updates on this? I have been experiencing the same issues for a bit and I bet that a repo with almost 4k stars has at least some people who know the solution to this. Any help is greatly appreciated.

**Comment by msqar** [comment]
OMG i thought it was my app, but yes... indeed the image quality is so lame after cropping. What's going on with this?

**Comment by msqar** [comment]
After reading some guy on a different bug ticket, and reading the docs... it says you need to set height and width. It's used for the cropping option as the final resolution. Depending on how high you set it there, the better the quality after cropping.

**Comment by cporawat** [comment]
Anyone found solution on this? I set more width/height and problem still remain.

**Comment by imbudhiraja** [comment]
@cporawat No luck buddy.

**Comment by jordangrant** [comment]
I saw poor image quality using 400x500, when I upped it to 1200x1500 the quality greatly improved. I'd say the solution is to set a greater height/width.

**Comment by leviduan** [comment]
> I saw poor image quality using 400x500, when I upped it to 1200x1500 the quality greatly improved. I'd say the solution is to set a greater height/width.

If you do that, you crop the picture, maybe it's not return the correct size

**Comment by axmedbek** [comment]
any solution ?

**Comment by hackrx** [comment]
I am also facing same problem but if we set cropping to false, then image quality is not decreased.
Please help if somebody found any solution

**Comment by hackrx** [comment]
This issue is solved, here is the code
`ImagePicker.openPicker({
      width: 1000,
      height: 800,
      cropping: true,
      includeBase64: false,
      compressImageQuality: 1,
      // compressImageMaxHeight: 1,
      // compressImageMaxWidth: 1,
      compressImageMaxWidth: 1500,
      compressImageMaxHeight: 1000,
    })`
We need to use height and width as max as possible, before this I was setting 
`ImagePicker.openPicker({
      width: 300,
      height: 300,
      cropping: true,
      includeBase64: false,
      compressImageQuality: 1,
      // compressImageMaxHeight: 1,
      // compressImageMaxWidth: 1,
      compressImageMaxWidth: 1500,
      compressImageMaxHeight: 1000,
    })`
And as a result, I was getting poor quality cropped image.

**Comment by MariusCatanoiu** [comment]
> width: 1000, height: 800

Is fine that you found a workaround. But this is not solved. The image will have a bigger space on the server.

**Comment by hackrx** [comment]
> > width: 1000, height: 800
> 
> Is fine that you found a workaround. But this is not solved. The image will have a bigger space on the server.

No, I have checked my firebase storage and I found that 4.2Mb image file is get converted to 478kb and this is what I want

**Comment by jdcaacbay-wp** [comment]
Still no fix for this?

**Comment by salik-a** [comment]
> I saw poor image quality using 400x500, when I upped it to 1200x1500 the quality greatly improved. I'd say the solution is to set a greater height/width.

this is worked for me bro, thanks a lot

**Comment by meznaric** [comment]
Applying the changes from [this PR](https://github.com/ivpusic/react-native-image-crop-picker/pull/1500) manually solves the issue for me.

iOS image is resized to 200x200 by default, not to selected crop rect.

**Comment by thangflydino** [comment]
same problem

**Comment by Dakuan** [comment]
+ 1 to cropping to aspect ratio rather than outright size. we run a photo app, we need to offer the ability to crop to common aspect ratios (eg 16:9, 4:3, 3:2) but *dont* want the images to be crushed.

**Comment by altsanz** [comment]
In our team we need the cropping to aspect ratio rather than outright size. How can we help to move mentioned PR forward?

**Comment by lucksp** [comment]
also looking for solution here.  My input is `{"height": 681, "width": 1689}`, and my output is `{"height": 80, "width": 200}`.  The `"cropRect": {"height": 275, "width": 681,}` is suggesting that it's using the actual size of the device screen?

**Comment by ajay-mandaviya** [comment]
Still facing same issue, if  u capture the document image or image which contain text 
After capture the images text is not visible clearly

**Comment by Kengin93** [comment]
With cropping:           
    ```
            ImagePicker.openPicker({
                    mediaType: 'photo',
                    cropping: true,
                    includeBase64: false,
                    freeStyleCropEnabled: true,
                    width: 1500, 
                    height: 1000,
                    compressImageQuality: 1,
                    compressImageMaxWidth: 1500, 
                    compressImageMaxHeight: 1000, 
                })
                    .then(async image => {
                        const { path } = image;
                        const file = await RNFS.readFile(path, 'base64');
                        const uploadingFile = new ReactNativeFile({
                            uri: path,
                            name: path.split('/').pop(),
                            type: 'image/jpeg',
                            data: file,
                        });
                        setSelectedFiles([uploadingFile]);
                    })
                    .catch(error => {
                        console.error(error);
                        setModalVisible(true);
                    });
                    
```

![before](https://github.com/user-attachments/assets/bfbd8c87-6756-44dc-bd0a-65f1aedd3e47)
![after](https://github.com/user-attachments/assets/f2498a17-0b52-45a0-9d4f-ff51eeb5fb68)
